aiohttp/streams.py

```diff
@@ -300,34 +300,41 @@ async def _wait(self, func_name: str) -> None:
             self._waiter = None
 
     async def readline(self) -> bytes:
+        return await self.readuntil()
+
+    async def readuntil(self, separator: bytes=b'\n') -> bytes:
+        seplen = len(separator)
+        if seplen == 0:
+            raise ValueError('Separator should be at least one-byte string')
+
         if self._exception is not None:
             raise self._exception
 
-        line = []
-        line_size = 0
+        chunk = b''
+        chunk_size = 0
         not_enough = True
 
         while not_enough:
             while self._buffer and not_enough:
                 offset = self._buffer_offset
-                ichar = self._buffer[0].find(b'\n', offset) + 1
-                # Read from current offset to found b'\n' or to the end.
+                ichar = self._buffer[0].find(separator, offset) + 1
```

Comment:
asyncio has much more effective solution. Why not to copy-paste it ?

Reply:
I did not want to change much the code.

What do you think should be copied from asyncio implementation?

Reply:
I think sync the implementation with asyncio would be great!
asyncio doesn't incrementally copy data to the buffer in a loop but calculates boundaries and move bytes once.

Reply:
Okay, I will fix it